guide/creating-your-bot/parsing-mention-arguments.md

```diff
@@ -0,0 +1,219 @@
+But what happens if you try to use the command like this?
+
+```
+!ban @Offender Because he was rude to @Victim
+```
+
+You might expect it to ban @Offender, because that is who you mentioned first.
+However, the Discord API does not send the mentions in the order they appear; They are sorted by their ID instead.
+
+If the @Victim happens to have joined Discord before @Offender and thus has a smaller ID he might get banned instead.  
+Or maybe someone uses a command incorrectly, the bot might still accept it but it will create an unexpected outcome.  
+Say someone accidentally used the ban command like this:
+
+```
+!ban Because he was rude to @Victim
+```
+
+The bot will still ban someone, but it will be the @Victim again. `message.mentions.users` still contains a mention, which the bot will use. But in reality you would want your bot to be able to tell the user he used the command incorrectly.
+
+### How Discord mentions work
+
+Discord uses a special syntax to embed mentions in a message. For user mentions it is the user's ID with `<@` at the start and `>` at the end, like this: `<@86890631690977280>`. If they have a nickname there will also be a be a `!` after the `@`.  
+Role mentions and channel mentions work similarly. Role mentions look like `<@&134362454976102401>` and channel mentions like `<#222197033908436994>`.
+
+That means when you receive a message from the Discord API and it contains mentions the message's content will contain that special syntax.  
+If you send
+
+<discord-messages>
+	<discord-message author="User" avatar="djs">
+		I think we should add @GoodPerson to the @Mod role.
+	</discord-message>
+</discord-messages>
+
+then the `message.content` for that message will look something like this
+
+<!-- eslint-skip -->
+```js
+'I think we should add <@86890631690977280> to the <@&134362454976102401> role.'
+```
+
+### Implementation
+
+So, how do you actually use this new information for your bot?  
+Most of your code will not change, however instead of using `message.mentions` to find the mentioned users you will have to do it manually.  
+This may sound scary at first, but once you see the code you will see it is actually pretty simple.
+
+Say you already have a simple command handler like this:
+
+```js
+client.on('message', message => {
+	// Don not go any further if there is no command prefix.
+	if (!message.content.startsWith(config.prefix)) return;
+
+	// Remove the prefix from the message.
+	const withoutPrefix = message.content.slice(config.prefix.length);
+	// Split the message by spaces
+	const split = withoutPrefix.split(/ +/);
+	// The first element is the command, the rest are the arguments.
+	const command = split[0];
+	const args = split.slice(1);
+
+	// ....Some commands here....
+});
+```
+
+Now you can easily test the waters by upgrading the avatar command from last time.
```

Comment:
"from last time" might be a bit unclear to those randomly landing on this page from somewhere else (e.g. a tag in the d.js server), so maybe a link linking to the previous command implementation would be nice.